CriteoPublisherSdk/Sources/Feedback/CR_FeedbackController.m

```diff
@@ -111,18 +113,36 @@ - (void)onBidConsumed:(CR_CdbBid *)consumedBid {
 }
 
 - (void)sendFeedbackBatch {
-  NSArray<CR_FeedbackMessage *> *feedbackMessages = [self.feedbackStorage popMessagesToSend];
-  if (feedbackMessages.count == 0) {
+  NSArray<CR_FeedbackMessage *> *messages = [self.feedbackStorage popMessagesToSend];
+  if (messages.count == 0) {
     return;
   }
 
+  NSDictionary<NSNumber *, NSArray<CR_FeedbackMessage *> *> *messagesByProfileId =
+      [messages cr_groupByKey:^NSNumber *(CR_FeedbackMessage *message) {
+        return message.profileId ?: @(CR_IntegrationFallback);
+      }];
+  [messagesByProfileId
+      enumerateKeysAndObjectsUsingBlock:^(NSNumber *profileId,
+                                          NSArray<CR_FeedbackMessage *> *messages, BOOL *stop) {
+        [self sendFeedbacks:messages profileId:profileId];
+      }];
+}
+
+- (void)sendFeedbacks:(NSArray<CR_FeedbackMessage *> *)messages profileId:(NSNumber *)profileId {
+  if (messages.count == 0) {
```

Comment:
Is this condition possible ?
Maybe even without both conditions (this one and the one at line 117), this is working fine as you will loop on an empty dictionnary

Reply:
Moving that precondition to the underlying ApiHandler method, where it makes sense.